File: backend/models/company_model.js
const mongoose = require('mongoose'); 
const db = require('../configuration/db');
const bcrypt = require('bcrypt');

const { Schema } = mongoose;

const companySchema = new Schema({
    email:{
        type:String,
        lowercase:true,
        required : true,
        unique: true,
    }, 
    name:{
        type:String,
        required : true,
    },      
    department:{
        type:String,
        required : true,
    },
    studentsApplied:{
        type:[Number],
        required : false,
    },

    
});

const companyModel = db.model('company', companySchema);

module.exports = companyModel;

File: frontend/lib/models/company_model.dart
class Company {
  final String email;
  final String nameCompany;
  final List<String> department;
  final List<int> studentsApplied;
  final List<int> studentsSelected;

  Company({
    required this.studentsSelected,
    required this.email,
    required this.nameCompany,
    required this.department,
    required this.studentsApplied,
  });

  // Factory method to create a Company object from the JSON data
  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      email: json['email'],
      nameCompany: json['name'],
      department: List<String>.from(json['department']),
      studentsApplied: List<int>.from(json['studentsApplied']),
      studentsSelected: List<int>.from(json['studentsSelected']),
    );
  }
}
